### DISCUSSION TITLE
A way you could run user code with root access

### DISCUSSION THREAD

**Discussion by hello-smile6** [raxod502/riju#152] [Ideas]
You might be able to use https://github.com/copy/v86 to run scripts with root access by either:

- Running VMs server-side and allowing serial access
- Running VMs client-side

Just thought I'd tell you about v86, because I really like your website.

**Comment by raxod502** [comment]
Sure, v86 works fine, but it has hugely slower performance than running native code on the server side. Root access can be granted with the current architecture (you still can't escape a Docker container with root access inside the container, except by finding a bug in Docker), but there's not much you can do with root access that you can't do without root access. Is there any particular use case that would be advantaged by root access on Riju?

**Reply by hello-smile6** [reply]
> Sure, v86 works fine, but it has hugely slower performance than running native code on the server side. Root access can be granted with the current architecture (you still can't escape a Docker container with root access inside the container, except by finding a bug in Docker), but there's not much you can do with root access that you can't do without root access. Is there any particular use case that would be advantaged by root access on Riju?

I'd really like to have something similar to Google's "Cloud Shell" thing that is open-source and would let me run a v86 networking proxy, or at least a WebSocket server.

**Reply by raxod502** [reply]
Well, that's out of scope for Riju, but if you do start a project like that, let me know---I'd love to check it out!

**Reply by hello-smile6** [reply]
> Well, that's out of scope for Riju, but if you do start a project like that, let me know---I'd love to check it out!

I'd consider it, but don't have the resources. Could you at least run a v86 Arch VM using Docker on your server and add it as an option for a language?

**Reply by raxod502** [reply]
No, sorry. I publish [financial information](https://github.com/raxod502/riju/tree/master/financials) about Riju's operations, and there is no spare budget for anything that's not directly related to the core goals of the project.

Even if there were budget, I would be hesitant to add something like this, because increasing the scope of what's covered on Riju outside of its original purpose will significantly increase the maintenance burden.

**Reply by hello-smile6** [reply]
> No, sorry. I publish [financial information](https://github.com/raxod502/riju/tree/master/financials) about Riju's operations, and there is no spare budget for anything that's not directly related to the core goals of the project.
> 
> Even if there were budget, I would be hesitant to add something like this, because increasing the scope of what's covered on Riju outside of its original purpose will significantly increase the maintenance burden.

Oh.

**Reply by hello-smile6** [reply]
> No, sorry. I publish [financial information](https://github.com/raxod502/riju/tree/master/financials) about Riju's operations, and there is no spare budget for anything that's not directly related to the core goals of the project.
> 
> Even if there were budget, I would be hesitant to add something like this, because increasing the scope of what's covered on Riju outside of its original purpose will significantly increase the maintenance burden.

Wow, you really need funding.
